src/lib/ast.ml

```diff
@@ -35,10 +84,38 @@ type t = CommandGroup.t list
 
 let to_list cg = cg
 
+let find_matching_datafile datafile_map path =
+  match List.assoc_opt path datafile_map with
+  | Some p -> Some p
+  | None ->
+      (* No full match, but can we find a prefix dir *)
+      List.fold_left
+        (fun acc i ->
```

Comment:
More idiomatic (read please ignore if you want ^^) to pull out the `(ipath, df)` here

```suggestion
        (fun acc (ipath, df) ->
```

Reply:
I thought I could do that, had tried it, and failed. I suspect that what happened was I had an error somewhere else, that I then accidentally cleared up when I moved to this form, and so I got a false positive on this pattern being the right way. Thanks!